File: vipclaw-common-entity/src/main/java/com/vipamp/vipclaw/common/entity/ProcessLogEntity.java
package com.vipamp.vipclaw.common.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 处理日志实体类
 *
 * @author vipamp
 * @since 2026-04-12
 */
@Data
@EqualsAndHashCode(callSuper = false)
@TableName("process_log")
@Schema(description = "处理日志实体类")
public class ProcessLogEntity implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * ID
     */
    @TableId(value = "id", type = IdType.AUTO)
    @Schema(description = "ID")
    private Long id;

    /**
     * 智能体 ID
     */
    @Schema(description = "智能体 ID")
    private Long agentId;

    /**
     * 智能体名称
     */
    @Schema(description = "智能体名称")
    private String agentName;

    /**
     * 会话 ID
     */
    @Schema(description = "会话 ID")
    private String sessionId;

    /**
     * 日志消息
     */
    @Schema(description = "日志消息")
    private String message;

    /**
     * 日志类型 (INFO/WARN/ERROR)
     */
    @Schema(description = "日志类型 (INFO/WARN/ERROR)")
    private String logType;

    /**
     * 异常堆栈信息
     */
    @Schema(description = "异常堆栈信息")
    private String stackTrace;

    /**
     * 时间戳
     */
    @Schema(description = "时间戳")
    private LocalDateTime ts;
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/admin/entity/ProcessLogEntity.kt
package com.vipamp.vipclaw.admin.entity

import io.swagger.v3.oas.annotations.media.Schema
import java.io.Serializable
import java.time.LocalDateTime

@Schema(description = "Process log entity")
class ProcessLogEntity : Serializable {
    companion object {
        private const val serialVersionUID = 1L
    }

    @Schema(description = "ID")
    var id: Long = 0

    @Schema(description = "Agent ID")
    var agentId: Long = 0

    @Schema(description = "Agent name")
    var agentName: String = ""

    @Schema(description = "Session ID")
    var sessionId: String = ""

    @Schema(description = "Log message")
    var message: String = ""

    @Schema(description = "Log type (INFO/WARN/ERROR)")
    var logType: String = "INFO"

    @Schema(description = "Exception stack trace")
    var stackTrace: String = ""

    @Schema(description = "Timestamp")
    var ts: LocalDateTime = LocalDateTime.now()
}
